sdk/core/Azure.Core/src/Shared/ARMChallengeAuthenticationPolicy.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace Azure.Core.Pipeline
+{
+    /// <summary>
+    /// A policy that sends an <see cref="AccessToken"/> provided by a <see cref="TokenCredential"/> as an Authentication header.
+    /// Note: This class is currently in preview and is therefore subject to possible future breaking changes.
+    /// </summary>
+    internal class ARMChallengeAuthenticationPolicy : BearerTokenChallengeAuthenticationPolicy
```

Comment:
We should follow up with the management plane team to see if this should ultimately live in Azure.Core shared source, or if they have a shared library which would be a better home.

Reply:
To be clear I'm fine with this living here for the moment, I'm just saying long term it might not be in Azure.Core.